op-node/rollup/derive/channel.go

```diff
@@ -157,6 +159,12 @@ func BatchReader(r io.Reader, l1InclusionBlock eth.L1BlockRef) (func() (BatchWit
 			L1InclusionBlock: l1InclusionBlock,
 		}
 		err := rlpReader.Decode(&ret.Batch)
-		return ret, err
+		if err != nil {
+			return ret, err
+		}
+		if ret.Batch.BatchType == SpanBatchType && !cfg.IsSpanBatch(ret.L1InclusionBlock.Time) {
```

Comment:
Have we specified how activation is done in the specs?

Reply:
In the implementation there's a placeholder span-batch-specific activation timestamp in the configuration, but with specs and final implementation I believe we can merge it with the Canyon hardfork definition.